src/steps/os/macos.rs

```diff
@@ -93,3 +94,121 @@ pub fn run_sparkle(ctx: &ExecutionContext) -> Result<()> {
     }
     Ok(())
 }
+
+pub fn update_xcodes(ctx: &ExecutionContext) -> Result<()> {
+    let xcodes = require("xcodes")?;
+    print_separator("Xcodes");
+
+    let should_ask = !(ctx.config().yes(Step::Xcodes) || ctx.config().dry_run());
+
+    let releases = ctx
+        .run_type()
+        .execute(&xcodes)
+        .args(["update"])
+        .output_checked_utf8()?
+        .stdout;
+
+    let releases_installed: Vec<String> = releases
+        .lines()
+        .filter(|r| r.contains("(Installed)"))
+        .map(String::from)
+        .collect();
+
+    if releases_installed.is_empty() {
+        println!("No Xcode releases installed.");
+        return Ok(());
+    }
+
+    let (installed_gm, installed_beta, installed_regular) =
+        releases_installed
+            .iter()
+            .fold((false, false, false), |(gm, beta, regular), release| {
+                (
+                    gm || release.contains("GM"),
+                    beta || release.contains("Beta"),
+                    regular || !(release.contains("GM") || release.contains("Beta")),
+                )
+            });
+
+    let releases_gm = releases
+        .lines()
+        .filter(|&r| r.matches("GM").count() > 0)
+        .map(String::from)
+        .collect();
+    let releases_beta = releases
+        .lines()
+        .filter(|&r| r.matches("Beta").count() > 0)
+        .map(String::from)
+        .collect();
+    let releases_regular = releases
+        .lines()
+        .filter(|&r| r.matches("GM").count() == 0 && r.matches("Beta").count() == 0)
+        .map(String::from)
+        .collect();
+
+    if installed_gm {
+        let _ = process_xcodes_releases(releases_gm, should_ask, ctx);
+    }
+    if installed_beta {
+        let _ = process_xcodes_releases(releases_beta, should_ask, ctx);
+    }
+    if installed_regular {
+        let _ = process_xcodes_releases(releases_regular, should_ask, ctx);
```

Comment:
Why are we ignoring the result of `process_xcodes_releases`, this could hide the possible errors

Reply:
That makes sense, I'll add an error checker. Should `process_xcodes_releases` return a boolean indicating success?